### PR TITLE
feat: update project dependencies

### PR DESCRIPTION
**Pull request by 1abhishekpandey**
## Description of the change

- Update react-native package to version `0.73.2`, using the official nx steps: https://nx.dev/features/automate-updating-dependencies.
- Fix the peer dependency issues (`npm WARN ERESOLVE overriding peer dependency`).
- For all the React Native packages: 
       - Manually replace the old (or outdated) package `metro-react-native-babel-preset` with `@react-native/babel-preset`. Refer here: https://reactnative.dev/blog/2023/12/06/0.73-debugging-improvements-stable-symlinks#babel-package-renames.
       - NX replaced `@nx/linter:eslint` with `@nx/eslint:lint`.
- Clean up unrequired changes in the `example/project.json`.
- Changes in the sample Android and iOS apps: Deleted the existing Android and iOS sample apps (in the `apps` folder). Then created new Android and iOS sample apps and configured everything over there.
- CI/CD: Update Java version to 17, as nx now supports this.
- Setup `distFolderPath` for all React Native packages. This is required in the `nx deploy ...` command.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix https://github.com/rudderlabs/rudder-sdk-react-native/pull/296 and https://github.com/rudderlabs/rudder-sdk-react-native/pull/295 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by saikumarrs — DISMISSED** [review]

**Review by ChryssaAliferi — APPROVED** [review]
I am approving the changes of this PR, however I do feel that the title is far from descriptive regarding the changes that are actually made here. This PR contains updates in the toolchain and project dependencies, apart from the package update to 0.73.2. In such cases, we handle those as two (or more) different tickets or PRs.  In our case now a change in the PR title is enough to portray the actual work made.

File: apps/example/android/build.gradle [hunk 1]
```diff
@@ -1,15 +1,11 @@
...
-        compileSdkVersion = 33
-        targetSdkVersion = 33
-        kotlinVersion = "1.8.20"
-
-        // We use NDK 23 which has both M1 support and is the side-by-side NDK version from AGP.
-        ndkVersion = "23.1.7779620"
+        compileSdkVersion = 34
+        targetSdkVersion = 34
+        ndkVersion = "25.1.8937393"
+        kotlinVersion = "1.8.0"
```
[the remark this answers is not in the document]
Comment (1abhishekpandey) [reply apps/example/android/build.gradle:8]: This `kotlinVersion` is set by the `nx` itself.
Should I update it to `1.8.22` or `1.9.23`?

**Comment by 1abhishekpandey** [comment]
> I am approving the changes of this PR, however I do feel that the title is far from descriptive regarding the changes that are actually made here. This PR contains updates in the toolchain and project dependencies, apart from the package update to 0.73.2. In such cases, we handle those as two (or more) different tickets or PRs. In our case now a change in the PR title is enough to portray the actual work made.

Does this PR title look good: `feat: update react native to v0.73.2, resolve peer dependencies, and enhance configuration`.

For your information: I've pushed all the changes as part of a single PR, as after following the official steps to update the React Native packages to version 0.73.2, I encountered multiple issues, including script failures and build issues in iOS. I made attempts to address these issues to prevent CI/CD actions from failing.

Please advise me on the best approach for handling such situations in the future. I'm eager to learn and apply the recommended practices going forward.

**Comment by ChryssaAliferi** [comment]
> > I am approving the changes of this PR, however I do feel that the title is far from descriptive regarding the changes that are actually made here. This PR contains updates in the toolchain and project dependencies, apart from the package update to 0.73.2. In such cases, we handle those as two (or more) different tickets or PRs. In our case now a change in the PR title is enough to portray the actual work made.
> 
> @ChryssaAliferi Does this PR title look good: `feat: update react native to v0.73.2, resolve peer dependencies, and enhance configuration`.
> 
> For your information: I've pushed all the changes as part of a single PR, as after following the official steps to update the React Native packages to version 0.73.2, I encountered multiple issues, including script failures and build issues in iOS. I made attempts to address these issues to prevent CI/CD actions from failing.
> 
> Please advise me on the best approach for handling such situations in the future. I'm eager to learn and apply the recommended practices going forward.

I believe `chore: update project dependencies` would be great! And we can discuss the rest offline! Thank you very much for the whole effort!!

**Review by ChryssaAliferi — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]

**Pull request merged**